src/laser_core/sortedqueue.py

```diff
@@ -0,0 +1,194 @@
+"""SortedQueue implementation using NumPy and Numba."""
+
+from typing import Any
+
+import numba as nb
+import numpy as np
+
+
+class SortedQueue:
+    """
+    A sorted (priority) queue implemented using NumPy arrays and sped-up with Numba.
+    Using the algorithm from the Python heapq module.
+    __init__ with an existing array of sorting values
+    __push__ with an index into sorting values
+    __pop__ returns the index of the lowest sorting value and its value
+    """
+
+    # https://github.com/python/cpython/blob/5592399313c963c110280a7c98de974889e1d353/Modules/_heapqmodule.c
+    # https://github.com/python/cpython/blob/5592399313c963c110280a7c98de974889e1d353/Lib/heapq.py
+
+    def __init__(self, capacity: int, values: np.ndarray):
+        self.indices = np.zeros(capacity, dtype=np.uint32)
```

Comment:
Given the particular nature of this class, I'd like more of a docstring on the constructor here.